app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountPresenter.java

```diff
@@ -0,0 +1,239 @@
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(notification -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void checkIfStoreIsInvalidAndRefresh() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(lifecycleEvent -> accountManager.accountStatus())
+        .filter(account -> !storeExistsInAccount(account))
+        .flatMap(account -> view.getStore()
+            .observeOn(scheduler)
+            .map(store -> store.getNodes()
+                .getMeta()
+                .getData())
+            .doOnNext(store -> view.refreshUI(store)))
+        .flatMap(__ -> accountManager.updateAccount()
+            .toObservable())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(account -> {
+        }, throwable -> CrashReport.getInstance()
+            .log(throwable));
+  }
+
+  private void handleLogOutClick() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(resumed -> signOutClick())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(signOutClick -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void handleSettingsClicked() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.settingsClicked())
+        .doOnNext(__ -> newAccountNavigator.navigateToSettings())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void handleNotificationHistoryClicked() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.notificationsClicked())
+        .doOnNext(__ -> newAccountNavigator.navigateToNotificationHistory())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private Observable<Void> signOutClick() {
+    return view.signOutClick()
+        .flatMap(click -> accountManager.logout()
+            .observeOn(scheduler)
+            .doOnCompleted(() -> {
+              resetAddressBookValues();
+              view.showLoginAccountDisplayable();
+            })
+            .doOnError(throwable -> crashReport.log(throwable)).<Void>toObservable())
+        .retry();
+  }
+
+  private void resetAddressBookValues() {
+    sharedPreferences.edit()
+        .putBoolean(ManagedKeys.ADDRESS_BOOK_SYNC, false)
+        .putBoolean(ManagedKeys.TWITTER_SYNC, false)
+        .putBoolean(ManagedKeys.FACEBOOK_SYNC, false)
+        .apply();
+  }
+
+  private boolean storeExistsInAccount(Account account) {
+    return account.getStore()
+        .getId() != 0;
+  }
+}
```

Comment:
this presenter is missing some observeOn's on the main thread

Reply:
fixed